### ISSUE TITLE
onAnimationEnd is not called after the opening animation (when using the `open` prop)

### ISSUE THREAD

**Issue by clementoriol** [emilkowalski/vaul#520]
**Issue**
On a controlled drawer (that uses the `open` prop to open/close, as its state is managed by a parent component),
the `onAnimationEnd` prop function is not called after the opening transition.

**Expectations**
`onAnimationEnd` should be *always* be called after the opening transition, even when using the `open` prop instead of a `Vaul.Trigger`

**Reproduction**
Codesandbox link: https://codesandbox.io/p/devbox/zzfxtc

**Comment by kavika-1** [comment]
Agreed - and would also be helpful if `onAnimationEnd` would also be called when the `open` prop is set to `false` to allow signaling the parent to wait for the close animation to complete before cleaning up the Drawer.

**Comment by travikk** [comment]
@emilkowalski facing this issue as well.

**Comment by MaximKondratev** [comment]
It is related to the same `onOpenChange` not being called as I described in my latest [issue](https://github.com/emilkowalski/vaul/issues/540) (my debug adventure is also described there and it is the root problem).  `onAnimationEnd` is called inside [onChange](https://github.com/emilkowalski/vaul/blob/73d06cdd3fd990bf4b83214cfe240c246908af0d/src/index.tsx#L173) (same as `onOpenChange`) which at the end expected to be triggered by Radix Dialog [here](https://github.com/emilkowalski/vaul/blob/73d06cdd3fd990bf4b83214cfe240c246908af0d/src/index.tsx#L759), but it does not for controlled state change.

I think the best way is either to move [onChange body](https://github.com/emilkowalski/vaul/blob/73d06cdd3fd990bf4b83214cfe240c246908af0d/src/index.tsx#L173) to useEffect to remove dependency of Radix implementation, or add useEffect inside useControllableState to call onChange there when state changes no matter controlled or not

**Comment by tgdn** [comment]
Facing the same issue. `onAnimationEnd` is never triggered when either opening or closing a controlled drawer.

IMO this is a bug because regardless of whether the drawer is controlled or not there is an animation opening/closing it.

Use case: reseting the drawer's state once the animation has ended to prevent UI changes to be visible to the user.

Right now the hacky way to do it is to use a timeout with `setTimeout`. But since it's hard to time the duration of the animation, the state could be reset while the animation is ongoing.

**Comment by tylerbecks** [comment]
I also have this issue.
`"vaul": "^1.1.2"`

**Comment by abuu-u** [comment]
`onAnimationEnd` on `Drawer.Content` worked for me
